indexer/sql/src/entity/transactions.rs

```diff
@@ -21,6 +42,49 @@ pub struct Model {
     pub gas_used: i64,
 }
 
+#[cfg(feature = "async-graphql")]
+#[ComplexObject]
+impl Model {
+    /// Nested Events from this transaction, from block on-disk caching
+    async fn nested_events(&self, ctx: &Context<'_>) -> Result<Option<String>> {
+        let (_, outcome) = load_tx_from_file(self, ctx).await?;
+
+        Ok(Some(outcome.events.to_json_string()?))
+    }
+
+    /// Flatten events from the indexer
+    async fn flatten_events(&self, ctx: &Context<'_>) -> Result<Vec<super::events::Model>> {
+        let loader = ctx.data_unchecked::<DataLoader<TransactionEventsDataLoader>>();
+        Ok(loader.load_one(self.clone()).await?.unwrap_or_default())
+    }
+
+    async fn messages(&self, ctx: &Context<'_>) -> Result<Vec<super::messages::Model>> {
+        let loader = ctx.data_unchecked::<DataLoader<TransactionMessagesDataLoader>>();
+        Ok(loader.load_one(self.clone()).await?.unwrap_or_default())
+    }
+
+    // async fn data(&self, ctx: &Context<'_>) -> Result<String> {
+    //     let (tx, _) = load_tx_from_file(self, ctx).await?;
+    //     Ok(tx.data.to_json_string()?)
+    // }
+
+    // async fn credential(&self, ctx: &Context<'_>) -> Result<String> {
+    //     let (tx, _) = load_tx_from_file(self, ctx).await?;
+    //     Ok(tx.credential.to_json_string()?)
+    // }
```

Comment:
Can these be deleted?

Reply:
I think they can yes, @j0nl1 it means `transaction.data` and `transaction.credential` in GraphQL if you used it won't be strings anymore but JSON objects (which is nice).